src/index.ts

```diff
@@ -33,14 +35,20 @@ export default function wasm(): Plugin {
       // Make a call to Vite's internal `fileToUrl` function by calling Vite's original WASM plugin's load()
       const originalLoadResult = (await originalWasmPlugin.load.call(this, id + "?init")) as string;
       const url = JSON.parse(/".+"/g.exec(originalLoadResult.trim().split("\n")[1])[0]) as string;
+      const importUrls = await Promise.all(
+        imports.map(async ({ from }) => {
+          const importerPath = id.slice(0, id.lastIndexOf("/")) + from.slice(from.lastIndexOf("/"));
```

Comment:
I don't really understand this path concatenation for `importerPath`, what would happen if the WASM file imports a JS file that are not in the same directory? e.g. when `from` is `../../../a/b/c/xxx.js`.

Reply:
This just replace file name:

```
id: 'C:/t/test-wasm2/node_modules/@jsona/openapi/index_bg.wasm'
from: "./index_bg.js"
fromUrl: "C:/t/test-wasm2/node_modules/@jsona/openapi/index_bg.js"
```

Reply:
What if

```
id: 'C:/t/test-wasm2/node_modules/@jsona/openapi/a/b/c/d/e/index_bg.wasm'
from: "../../f/g/index_bg.js"
```

Reply:
The index_bg.wasm always point to index_bg.js. in same directory.

Reply:
It only appies to `wasm-bindgen`, you can't assume all WASM files are generated by `wasm-bindgen`. I don't see in the WASM ESM spec it has any restrictions on the imported files' directory.